# HG changeset patch
# User rupashka
# Date 1289665358 -10800
# Node ID d385b33c0db056aa722f93eed1970f70f0b8f119
# Parent  286b14273037644c62e7f263d1d9249f4308783b
6613904: javax.swing.GroupLayout.createParallelGroup(..) doesn't throw IllegalArgumentException for null arg
Reviewed-by: peterz

diff -r 286b14273037 -r d385b33c0db0 src/share/classes/javax/swing/GroupLayout.java
--- openjdk.orig/jdk/src/share/classes/javax/swing/GroupLayout.java	Sat Nov 13 13:04:47 2010 +0300
+++ openjdk/jdk/src/share/classes/javax/swing/GroupLayout.java	Sat Nov 13 19:22:38 2010 +0300
@@ -653,6 +653,10 @@
      */
     public ParallelGroup createParallelGroup(Alignment alignment,
             boolean resizable){
+        if (alignment == null) {
+            throw new IllegalArgumentException("alignment must be non null");
+        }
+
         if (alignment == Alignment.BASELINE) {
             return new BaselineGroup(resizable);
         }
diff -r 286b14273037 -r d385b33c0db0 test/javax/swing/GroupLayout/6613904/bug6613904.java
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ openjdk/jdk/test/javax/swing/GroupLayout/6613904/bug6613904.java	Sat Nov 13 19:22:38 2010 +0300
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2010, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 6613904
+ * @summary javax.swing.GroupLayout.createParallelGroup(..) doesn't throw IllegalArgumentException for null arg
+ * @author Pavel Porvatov
+ */
+
+import javax.swing.*;
+
+public class bug6613904 {
+    public static void main(String[] args) {
+        SwingUtilities.invokeLater(new Runnable() {
+            public void run() {
+                GroupLayout groupLayout = new GroupLayout(new JPanel());
+
+                try {
+                    groupLayout.createParallelGroup(null);
+
+                    throw new RuntimeException("groupLayout.createParallelGroup(null) doesn't throw IAE");
+                } catch (IllegalArgumentException e) {
+                    // Ok
+                }
+
+                try {
+                    groupLayout.createParallelGroup(null, true);
+
+                    throw new RuntimeException("groupLayout.createParallelGroup(null, true) doesn't throw IAE");
+                } catch (IllegalArgumentException e) {
+                    // Ok
+                }
+
+                try {
+                    groupLayout.createParallelGroup(null, false);
+
+                    throw new RuntimeException("groupLayout.createParallelGroup(null, false) doesn't throw IAE");
+                } catch (IllegalArgumentException e) {
+                    // Ok
+                }
+            }
+        });
+    }
+}
